File: ios/Packages/SolennixCore/Sources/SolennixCore/Models/ProductIngredient.swift
import Foundation

public struct ProductIngredient: Codable, Identifiable, Sendable, Hashable {
    public let id: String
    public let productId: String
    public let inventoryId: String
    public let quantityRequired: Double
    public var capacity: Double?
    public var bringToEvent: Bool?
    public let createdAt: String
    // Flattened inventory joined fields
    public var ingredientName: String?
    public var unit: String?
    public var unitCost: Double?
    public var type: InventoryType?

    enum CodingKeys: String, CodingKey {
        case id
        case productId = "product_id"
        case inventoryId = "inventory_id"
        case quantityRequired = "quantity_required"
        case capacity
        case bringToEvent = "bring_to_event"
        case createdAt = "created_at"
        case ingredientName = "ingredient_name"
        case unit
        case unitCost = "unit_cost"
        case type
    }
}

File: android/core/model/src/main/java/com/creapolis/solennix/core/model/ProductIngredient.kt
package com.creapolis.solennix.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ProductIngredient(
    val id: String,
    @SerialName("product_id") val productId: String,
    @SerialName("inventory_id") val inventoryId: String,
    @SerialName("quantity_required") val quantityRequired: Double,
    val capacity: Double? = null,
    @SerialName("bring_to_event") val bringToEvent: Boolean? = null,
    @SerialName("created_at") val createdAt: String,
    @SerialName("ingredient_name") val ingredientName: String? = null,
    val unit: String? = null,
    @SerialName("unit_cost") val unitCost: Double? = null,
    val type: InventoryType? = null
)
